### ISSUE TITLE
ImportError: cannot import name 'default_pg_timeout' from 'torch.distributed'

### ISSUE THREAD

**Issue by SreehariSankar** [Lightning-AI/lightning#16931]
### Bug description

Cannot import pytorch-lightning.

Mac OSX v12.3, M1 Pro Chip
Conda 22.9.0
Python 3.8.16
pytorch-lightning 1.9.0
torch 1.13.1
transformers 4.21.3
pytorch-ignite 0.4.10
py-cpuinfo 9.0.0

Error Stack provided in the error log section.
Thanks in advance!

### Error messages and logs

Stack: 
Cell In[3], line 7
      5 import warnings
      6 warnings.filterwarnings('ignore')
----> 7 import pytorch_lightning as pl
      8 import re, copy
      9 import numpy as np

File ~/anaconda3/envs/venv/lib/python3.8/site-packages/pytorch_lightning/__init__.py:35
     32     _logger.propagate = False
     34 from lightning_fabric.utilities.seed import seed_everything  # noqa: E402
---> 35 from pytorch_lightning.callbacks import Callback  # noqa: E402
     36 from pytorch_lightning.core import LightningDataModule, LightningModule  # noqa: E402
     37 from pytorch_lightning.trainer import Trainer  # noqa: E402

File ~/anaconda3/envs/venv/lib/python3.8/site-packages/pytorch_lightning/callbacks/__init__.py:28
     26 from pytorch_lightning.callbacks.prediction_writer import BasePredictionWriter
     27 from pytorch_lightning.callbacks.progress import ProgressBarBase, RichProgressBar, TQDMProgressBar
---> 28 from pytorch_lightning.callbacks.pruning import ModelPruning
     29 from pytorch_lightning.callbacks.quantization import QuantizationAwareTraining
     30 from pytorch_lightning.callbacks.rich_model_summary import RichModelSummary

File ~/anaconda3/envs/venv/lib/python3.8/site-packages/pytorch_lightning/callbacks/pruning.py:31
     29 import pytorch_lightning as pl
     30 from pytorch_lightning.callbacks.callback import Callback
---> 31 from pytorch_lightning.core.module import LightningModule
     32 from pytorch_lightning.utilities.exceptions import MisconfigurationException
     33 from pytorch_lightning.utilities.rank_zero import rank_zero_debug, rank_zero_only

File ~/anaconda3/envs/venv/lib/python3.8/site-packages/pytorch_lightning/core/__init__.py:16
      1 # Copyright The PyTorch Lightning team.
      2 #
      3 # Licensed under the Apache License, Version 2.0 (the "License");
   (...)
     12 # See the License for the specific language governing permissions and
     13 # limitations under the License.
     15 from pytorch_lightning.core.datamodule import LightningDataModule
---> 16 from pytorch_lightning.core.module import LightningModule
     18 __all__ = ["LightningDataModule", "LightningModule"]

File ~/anaconda3/envs/venv/lib/python3.8/site-packages/pytorch_lightning/core/module.py:34
     31 from torchmetrics import Metric, MetricCollection
     32 from typing_extensions import Literal
---> 34 import lightning_fabric as lf
     35 import pytorch_lightning as pl
     36 from lightning_fabric.loggers import Logger as FabricLogger

File ~/anaconda3/envs/venv/lib/python3.8/site-packages/lightning_fabric/__init__.py:23
     17 # In PyTorch 2.0+, setting this variable will force `torch.cuda.is_available()` and `torch.cuda.device_count()`
     18 # to use an NVML-based implementation that doesn't poison forks.
     19 # https://github.com/pytorch/pytorch/issues/83973
     20 os.environ["PYTORCH_NVML_BASED_CUDA_CHECK"] = "1"
---> 23 from lightning_fabric.fabric import Fabric  # noqa: E402
     24 from lightning_fabric.utilities.seed import seed_everything  # noqa: E402
     26 __all__ = ["Fabric", "seed_everything"]

File ~/anaconda3/envs/venv/lib/python3.8/site-packages/lightning_fabric/fabric.py:34
     32 from lightning_fabric.plugins import Precision  # avoid circular imports: # isort: split
     33 from lightning_fabric.accelerators.accelerator import Accelerator
---> 34 from lightning_fabric.connector import _Connector, _PLUGIN_INPUT, _PRECISION_INPUT
     35 from lightning_fabric.strategies import DeepSpeedStrategy, FSDPStrategy, SingleDeviceStrategy, Strategy, XLAStrategy
     36 from lightning_fabric.strategies.strategy import _Sharded, TBroadcast

File ~/anaconda3/envs/venv/lib/python3.8/site-packages/lightning_fabric/connector.py:45
     43 from lightning_fabric.plugins.precision.fsdp import FSDPPrecision
     44 from lightning_fabric.plugins.precision.precision import _PRECISION_INPUT, _PRECISION_INPUT_INT, _PRECISION_INPUT_STR
---> 45 from lightning_fabric.strategies import (
     46     DeepSpeedStrategy,
     47     SingleDeviceStrategy,
     48     SingleTPUStrategy,
     49     Strategy,
     50     STRATEGY_REGISTRY,
     51     XLAStrategy,
     52 )
     53 from lightning_fabric.strategies.ddp import _DDP_FORK_ALIASES
     54 from lightning_fabric.strategies.fsdp import _FSDP_ALIASES, FSDPStrategy

File ~/anaconda3/envs/venv/lib/python3.8/site-packages/lightning_fabric/strategies/__init__.py:17
     15 from lightning_fabric.strategies.deepspeed import DeepSpeedStrategy  # noqa: F401
     16 from lightning_fabric.strategies.dp import DataParallelStrategy  # noqa: F401
---> 17 from lightning_fabric.strategies.fsdp import FSDPStrategy  # noqa: F401
     18 from lightning_fabric.strategies.parallel import ParallelStrategy  # noqa: F401
     19 from lightning_fabric.strategies.registry import _call_register_strategies, _StrategyRegistry

File ~/anaconda3/envs/venv/lib/python3.8/site-packages/lightning_fabric/strategies/fsdp.py:21
     19 import torch
     20 from torch import Tensor
---> 21 from torch.distributed import default_pg_timeout
     22 from torch.nn import Module
     23 from torch.optim import Optimizer

ImportError: cannot import name 'default_pg_timeout' from 'torch.distributed' (/Users/sreeharisankar/anaconda3/envs/venv/lib/python3.8/site-packages/torch/distributed/__init__.py)

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

**Comment by adamjstewart** [comment]
Duplicate of #16637, which was fixed by #16658. It looks like 1.9.0 was the only version that contained this bug. If you upgrade to 1.9.1+ you should be fine.

**Comment by SreehariSankar** [comment]
Resolved. upgraded version works fine.
